mlir/lib/Rewrite/FrozenRewritePatternSet.cpp

```diff
@@ -73,7 +73,7 @@ FrozenRewritePatternSet::FrozenRewritePatternSet(
   // Functor used to walk all of the operations registered in the context. This
   // is useful for patterns that get applied to multiple operations, such as
   // interface and trait based patterns.
-  std::vector<RegisteredOperationName> opInfos;
+  SmallVector<RegisteredOperationName> opInfos;
```

Comment:
I'm not immediately seeing why this change is related to the rest of the PR (although probably a good change in general, I would expect)?

Reply:
Reverted this, since now we continue to use `ArrayRef` as the return type. I had initially changed the type to `SmallVector` which caused the old logic to throw an error for `=`.